### PR TITLE
Update fabric8 dependency minor version

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- version update

### Description
Update fabric8 kubernetes client to 5.5.0
Breaking changes:
- StatusUpdatable.updateStatus deprecated, please use patchStatus, editStatus, or replaceStatus
- Waitable.waitUntilReady and Waitable.waitUntilCondition with throw a KubernetesClientTimeoutException instead of an IllegalArgumentException

more info in [release notes](https://github.com/fabric8io/kubernetes-client/releases)

### Checklist
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Apart form the failing build it looks good.

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java [hunk 1]
```diff
@@ -233,13 +234,13 @@ public void testUpdateStatusAfterResourceUpdatedThrowsKubernetesClientException(
...
                     return op.updateStatusAsync(Reconciliation.DUMMY_RECONCILIATION, newStatus);
                 })
-                .onComplete(context.failing(e -> context.verify(() -> {
-                    assertThat("Exception was not KubernetesClientException, it was : " + e.toString(),
-                            e, instanceOf(KubernetesClientException.class));
-                    updateFailed.complete();
+                .onComplete(context.succeeding(res -> context.verify(() -> {
+                    assertThat(res.getMetadata().getName(), is(resourceName));
+                    assertThat(res.getMetadata().getNamespace(), is(namespace));
+                    updateStatus.complete();
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java:240]: Does this mean that it fixes #5326?

Comment (sknot-rh) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java:240]: I would need to elaborate. But when the resource & status is modified simultaneously, it does not throw.

Comment (scholzj) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java:240]: Well, the exception you removed in the test seems to be what #5326 complains about, or?

Comment (sknot-rh) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractCustomResourceOperatorIT.java:240]: Yes, you are right.

**Review by scholzj — APPROVED** [review]

**Comment by shawkins** [comment]
If you are using waitUntilReady/Condition 5.5 does have a related bug - https://github.com/fabric8io/kubernetes-client/issues/3318 means that it's possible to miss events on  a  relist (after http gone on the watch).  This is fixed in 5.6, which is being released today.

**Pull request merged**